oauth_sample/README.md

```diff
@@ -0,0 +1,141 @@
+# QPP OAuth2 Sample Client
+
+Node.js client application to demonstrate OAuth2 integration with [QPP](https://qpp.cms.gov).
+
+OAuth2 allows users to grant access to desired resources for third party applications,
+giving them the ability to enable and disable this access whenever they want.
+
+This client supports the [Authorization Code Flow](http://tools.ietf.org/html/draft-ietf-oauth-v2-31#section-4.1) and [Authorization Code Flow with PKCE](https://tools.ietf.org/html/rfc7636).
+
+## Requirements
+
+The Node client library is tested against Node 10 LTS.
+
+## Getting started
+
+### Installation
+
+Install the dependencies:
+
+```bash
+npm install
+```
+
+### Setup
+[Register your application](https://qpp.cms.gov/api/auth/docs/#/Oauth/post_api_auth_oauth_applications__resource_id_) with QPP. 
+
+The URIs specified when creating the application should reflect the `APP_BASE_URL` and `APP_PORT` values in the provided `config.js` file. By default, the application host is `http://localhost:3000`.
+
+Many of the fields required for registration may contain arbitrary "dummy" values, but the following *must* be configured correctly in order to use the sample client:
+
+* `redirectURIs`: **must** contain the absolute URI path to the `/logged_in` endpoint for your local instance of this application
+    * E.g. `http://localhost:3000/logged_in`
+* `applicationType`: ("web" | "native")
+    * `web` applications will receive both a client ID and a client secret
+    * `native` applications will receive only a client ID
+
+Upon registering the application, copy `config.js.example` to `config.js` and use the returned OAuth 2.0 credentials to populate the following values in `config.js`:
+
+  * `CLIENT_ID` - Service registered client id. Required.
+  * `CLIENT_SECRET` - Service registered client secret. Required for applications which can securely store a secret. Native or mobile applications will not have a client secret and will use the [PKCE Flow](https://auth0.com/docs/flows/concepts/auth-code-pkce) for authorization.
+
+### Run
+Start the example application with:
```

Comment:
I think you previously had a comment like, "Note: if you change any values in `config.js` you'll need to restart the application in order for them to take effect". I thought that was helpful so minor suggestion you bring it back :D

Reply:
I considered this and could have gone either way.  Your vote breaks my inner tie.  Adding it back :)